### PR TITLE
Canonical Bug fix

### PR DESCRIPTION
**Pull request by shadrackkibet**
@lilyclements  this is ready for review.Here i have sorted out the following;

- Display of graphs on the output window.This was producing an empty grid on the output window.

- Sorted out iposition problem. The display on the output window was not in order.

Issues:

- The two checkboxes on the sdg behaves funny. When one is checked the other one checks automatically on getting out of the sub.

- The data_name parameter BUG. This is an issue with the settingRcode in this dialog. It seems adding additional parameter pairs don't work properly. I don't understand why this happens! However i have added them manually for now. Have a look at this please.

### CODE REVIEW

**Comment by lilyclements** [comment]
@shadrackkibet, interesting finds! Looking at the code, all the changes look good - but I'll review this properly tomorrow if that's okay. Just in case I find any other little bugs and fix them.

That's a shame about the `bExcludeAssignedFunctionOutput = False`. Such an easy fix and so a shame we all missed that!

Interesting about the add additional parameter code - @AlexSananka reported he was having the same problem on PCA. I've not experienced this problem, can you explain to me how to replicate it? We should discuss it through with @dannyparsons

**Comment by shadrackkibet** [comment]
This is the same dialog that Alex was working on. The issue we were experiencing on PCA was quiet different from this. The issue of the axes parameter? hope remember.

File: instat/dlgCanonicalCorrelationAnalysis.vb [hunk 1]
```diff
@@ -88,27 +89,30 @@ Public Class dlgCanonicalCorrelationAnalysis
...
         clsRYCoefFunction.AddParameter("value1", Chr(34) & "ycoef" & Chr(34))
 
         ucrBase.clsRsyntax.ClearCodes()
-        ucrBase.clsRsyntax.AddToAfterCodes(clsRXCoefFunction, 1)
-        ucrBase.clsRsyntax.AddToAfterCodes(clsRCanCorFunction, 2)
+        ucrBase.clsRsyntax.AddToAfterCodes(clsRCanCorFunction, iPosition:=0)
+        ucrBase.clsRsyntax.AddToAfterCodes(clsRXCoefFunction, iPosition:=1)
+        ucrBase.clsRsyntax.AddToAfterCodes(clsRYCoefFunction, iPosition:=2)
         ucrBase.clsRsyntax.SetBaseRFunction(clsDefaultFunction)
+        DataName()
```
Comment (shadrackkibet) [inline instat/dlgCanonicalCorrelationAnalysis.vb]: @lilyclements  i have called this here to fix that problem meanwhile. If you remove this then use the Rcode, you will  experience the issue of data_name missing.

**Comment by lilyclements** [comment]
I didn't mean the axes issue. I meant on PCA we were getting a problem I believe with the data_name. I believe @AlexSananka was saying it wasn't being read in properly on the SetRCodes part?

**Comment by dannyparsons** [comment]
@shadrackkibet can you respond to this

**Comment by shadrackkibet** [comment]
This is ready for you to review.

**Comment by lilyclements** [comment]
Do we know why this data_name issue is occurring? I haven't been able to replicate it but I know @shadrackkibet and @AlexSananka have spoken about it both here and on PCA.

**Comment by shadrackkibet** [comment]
will skype you to explain this.

**Comment by AlexSananka** [comment]
@lilyclements Have you tested this?are you still having the same issue here?

**Comment by lilyclements** [comment]
I'm not experiencing this issue but you both have been reporting it before and said that's why you're calling it manually and hence commented out line 113-115.

I had a little play around and the issue is only occurring to the `AddAdditionalCodeParameterPair` classes which call `data_name`, not the one calling `SetRCode`

**Comment by shadrackkibet** [comment]
@lilyclements  anything here still remaining to address apart from the issue of AddAdditionalCodeParameterPair?

**Comment by lilyclements** [comment]
Nope that's the only problem I've seen here, although I haven't tested this rigorously since the last update so if I get time I can do that.

**Comment by maxwellfundi** [comment]
@lilyclements  can you check if things here are fixed?

**Comment by lilyclements** [comment]
The issue of AddAdditionalParameterPair needs to be either addressed or discussed still.

**Comment by Lunalo** [comment]
I think i will have a look at this with @shadrackkibet before he leaves

**Pull request closed without merging**